### DISCUSSION TITLE
Is it good to use "use client" in every page.tsx?

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#55142] [Help] [answered]
### Summary

Is it good to use "use client" in every page.tsx? Otherwise, it is difficult to work with redux and context. I once heard that this is bad practice, and page.tsx should be left to the server. Where is the truth?)

**Comment by acomanescu** [comment, marked as answer]
There isn't bad or good practice. It depends on your actual use case. I do prefer to stick with the default, "use server", and only transform into "use client" when there is something that requires that. I tend more to create separate components where I use "use client" instead of adding the flag to the page. You can add your redux/react context in the layout (segment or root), or in the page, then and use the data inside client components through hooks.
